src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs

```diff
@@ -0,0 +1,283 @@
+﻿using System.Text.Json;
+using Microsoft.TeamFoundation.WorkItemTracking.WebApi;
+using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi.Patch.Json;
+
+namespace Microsoft.OneFuzz.Service;
+
+public interface IAdo {
+    public Async.Task NotifyAdo(AdoTemplate config, Container container, string filename, IReport reportable, bool failTaskonTransientError);
+
+}
+
+public class Ado : NotificationsBase, IAdo {
+    public Ado(ILogTracer logTracer, IOnefuzzContext context) : base(logTracer, context) {
+    }
+
+    public async Async.Task NotifyAdo(AdoTemplate config, Container container, string filename, IReport reportable, bool failTaskonTransientError) {
+        if (reportable is RegressionReport) {
+            _logTracer.Info($"ado integration does not support regression report. container:{container} filename:{filename}");
+            return;
+        }
+
+        var report = (Report)reportable;
+
+        var notificationInfo = @$"job_id:{report.JobId} task_id:{report.TaskId}
+container:{container} filename:{filename}";
+
+        _logTracer.Info($"notify ado: {notificationInfo}");
+
+        try {
+            var ado = await AdoConnector.AdoConnectorCreator(_context, container, filename, config, report, _logTracer);
+            await ado.Process(notificationInfo);
+        } catch (Exception e) {
+            /*
+            TODO: Catch these
```

Comment:
TODO for later ?

Reply:
Yup, it's going to be part of testing ado